editor/src/components/canvas/controls/grid-controls-for-strategies.tsx

```diff
@@ -68,142 +92,174 @@ export type GridMeasurementHelperData = {
+    const rows = getCellsCount(containerGridProperties.gridTemplateRows)
+    const borderTopWidth = parseCSSLength(computedStyle.borderTopWidth)
+    const borderRightWidth = parseCSSLength(computedStyle.borderRightWidth)
+    const borderBottomWidth = parseCSSLength(computedStyle.borderBottomWidth)
+    const borderLeftWidth = parseCSSLength(computedStyle.borderLeftWidth)
+    const border: BorderWidths = {
+      top: isRight(borderTopWidth) ? borderTopWidth.value.value : 0,
+      right: isRight(borderRightWidth) ? borderRightWidth.value.value : 0,
+      bottom: isRight(borderBottomWidth) ? borderBottomWidth.value.value : 0,
+      left: isRight(borderLeftWidth) ? borderLeftWidth.value.value : 0,
+    }
+    const justifyContent = getFlexJustifyContent(computedStyle.justifyContent)
+    const alignContent = getAlignContent(computedStyle.alignContent)
+    const padding = defaultEither(
+      sides(undefined, undefined, undefined, undefined),
+      applicative4Either(
+        applicativeSidesPxTransform,
+        parseCSSLength(computedStyle.paddingTop),
+        parseCSSLength(computedStyle.paddingRight),
+        parseCSSLength(computedStyle.paddingBottom),
+        parseCSSLength(computedStyle.paddingLeft),
+      ),
+    )
+
+    const gap = defaultEither(
+      null,
+      mapEither((n) => n.value, parseCSSLength(computedStyle.gap)),
+    )
+
+    const rowGap = defaultEither(
+      null,
+      mapEither((n) => n.value, parseCSSLength(computedStyle.rowGap)),
+    )
+
+    const columnGap = defaultEither(
+      null,
+      mapEither((n) => n.value, parseCSSLength(computedStyle.columnGap)),
+    )
+
+    const elementRect = domRectToScaledCanvasRectangle(
+      boundingRectangle,
+      1 / scale,
+      getRoundingFn('nearest-half'),
+    )
+    const parentRect = domRectToScaledCanvasRectangle(
+      canvasRootContainer.getBoundingClientRect(),
+      1 / scale,
+      getRoundingFn('nearest-half'),
+    )
+    const frame = Utils.offsetRect(elementRect, Utils.negate(parentRect))
+
+    return {
+      frame: frame,
+      gridTemplateColumns: containerGridProperties.gridTemplateColumns,
+      gridTemplateRows: containerGridProperties.gridTemplateRows,
+      gridTemplateColumnsFromProps: containerGridPropertiesFromProps.gridTemplateColumns,
+      gridTemplateRowsFromProps: containerGridPropertiesFromProps.gridTemplateRows,
+      gap: gap,
+      rowGap: rowGap,
+      columnGap: columnGap,
+      justifyContent: justifyContent,
+      alignContent: alignContent,
+      padding: padding,
+      rows: rows,
+      columns: columns,
+      cells: columns * rows,
+      border: border,
+    }
+  }
+}
+
+export function useGridMeasurementHelperData(
+  elementPath: ElementPath,
+): GridMeasurementHelperData | undefined {
+  const scale = useEditorState(
+    Substores.canvas,
+    (store) => store.editor.canvas.scale,
+    'useGridMeasurementHelperData scale',
   )
+
+  useMonitorChangesToEditor()
```

Comment:
isn't useMonitorChangesToElements good enough here?
even if you call useKeepReferenceEqualityIfPossible on the return value, the component calling this hook will be forced to rerender by useMonitorChangesToEditor

Reply:
Without this, it meant anything slightly more funky like asynchronously updated values would leave the helpers out of date. Even something simple like a sibling of the grid expanding and moving the grid as a result would leave the helpers in a weird state.